docs/reference/builder.md

```diff
@@ -152,18 +152,25 @@ Example (parsed representation is displayed after the `#`):
     ADD . $foo       # ADD . /bar
     COPY \$foo /quux # COPY $foo /quux
 
-The instructions that handle environment variables in the `Dockerfile` are:
+Environment variables are supported by the following list of instructions in 
+the `Dockerfile`:
 
-* `ENV`
 * `ADD`
 * `COPY`
-* `WORKDIR`
+* `ENV`
 * `EXPOSE`
-* `VOLUME`
 * `USER`
+* `WORKDIR`
+* `VOLUME`
+
+as well as:
 
-`ONBUILD` instructions are **NOT** supported for environment replacement, even
-the instructions above.
+* `ONBUILD` (when combined with one of the supported instructions above)
+
+> **Note**:
+> prior to 1.4, `ONBUILD` instructions did **NOT** support environment 
+> variable, even when combined with any of the instructions listed above.
+> This limitation was later fixed.
```

Comment:
s/fixed/removed/ :smile:

Reading again, I think that line isn't needed at all, because it's already clear from "prior to 1.4", so I think it can be removed.